cpuid/src/bit_helper.rs

```diff
@@ -0,0 +1,188 @@
+/// Structure representing a range of bits in a number
+///
+pub struct BitRange {
+    /// most significant bit
+    pub left: u32,
+    /// least significant bit
+    pub right: u32,
+}
+
+pub trait BitHelper {
+    /// Changes the value of the bit at position `pos` to `val`
+    ///
+    fn set_bit(&mut self, pos: u32, val: bool) -> &mut Self;
+
+    /// Reads the value stored within the specified range of bits
+    ///
+    fn read_bits(&self, bit_range: &BitRange) -> Self;
+
+    /// Stores a value within the specified range of bits
+    ///
+    fn write_bits(&mut self, bit_range: &BitRange, val: Self) -> &mut Self;
+}
+
+fn get_mask(bit_range: &BitRange) -> u32 {
+    ((((1 as u64) << (bit_range.left - bit_range.right + 1)) - 1) << bit_range.right) as u32
+}
+
+impl BitHelper for u32 {
+    fn set_bit(&mut self, pos: u32, val: bool) -> &mut Self {
+        *self &= !(1 << pos);
+        *self |= (val as u32) << pos;
+        self
+    }
+
+    fn read_bits(&self, bit_range: &BitRange) -> Self {
+        (self & get_mask(&bit_range)) >> bit_range.right
```

Comment:
Would be nice to not allow bitranges that don't fit the `Self` type size.

Reply:
Done